lotemplate/classes.py

```diff
@@ -77,6 +78,31 @@ def restart(self) -> None:
         self.__init__(self.host, self.port)
 
 
+class IfStatement:
+    """
+    Class representing an if statement in a template libreoffice
+    """
+    start_regex = r'\[\s*if\s*\$(\w+(\(.+?\))?)\s*(\=\=|\!\=)\s*([\w\-\.\,]+)\s*\]'
```

Comment:
```py
start_regex = r'\[\s*if\s*\$(\w+)\s*(==|!=)\s*(.+(?<!\s))\s*\]'
```

Reply:
1. With this regex, I introduced the possibility to write text containing spaces, as well as the `]` character. But the capture group 3 used to match the trailing spaces, just like this:
  `[ if $test == test with spaces ]` -> group 3 matches `test with spaces ` instead of `test with spaces`.
  So, I added this negative lookbehind just after the text pattern  `.+`. It ensures that the last character of `.+` isn't a whitespace character (`\s`).
2. It doesn't supports recursion and matches function variables that are syntactically false. It leads to inconsistency within the program, where function variables behave differently following where they're written, and errors (as a variable function containing another variable function aren't matched in a condition bloc, which causes the block not to be replaced.
  With everything that has been said, I do believe that a "partial solution" isn't a solution at all. Variable functions should be supported in a condition bloc, or not, but shouldn't be "partially supported". Covering a part of cases isn't covering cases at all.

Reply:
We could use that regex to parse vars : `\$\w+\(((?:\\.|[^\\\(]|[^\\\)])*?)\)`

The only implication is to escape parenthesis inside the function vars. This has two big advantages for me : 
* it doesn't impose any syntax inside the function vars. 
* the regex works either in python (base re module) and libreoffice

with python : 
![image](https://user-images.githubusercontent.com/393066/224933898-317e1810-9728-4b39-8e6a-46a6dd7c6ca4.png)

with libreoffice : 
![image](https://user-images.githubusercontent.com/393066/224934063-563051a8-641f-43b2-a963-1ad8201c9e8b.png)

(the test sentence is `une variable $odoo($tutu\(ma-sub-var\) and tutu) outside $gloubi(super) var et encore 1 : $odoo($tutu\($tata\(gloubi\) * ma-sub-var\) and tutu) end of the sentence`)

Reply:
I simplified a bit the regex and I pushed the modified code (for vars parsing and if parsing) in the commit af7acc8

I also made a small hack to make the regex more readable in https://github.com/Probesys/lotemplate/pull/11/files#diff-a2c1cf905e598b45699bcc013d16ef58fe5bae902b82d42a376e40d7d668ecabR85